python/_tskitmodule.c

```diff
@@ -10774,6 +10775,62 @@ TreeSequence_make_array(TreeSequence *self, tsk_size_t size, int dtype, void *da
     return make_owned_array((PyObject *) self, size, dtype, data);
 }
 
+PyObject *
+TreeSequence_decode_ragged_string_column(
+    TreeSequence *self, tsk_size_t num_rows, const char *data, const tsk_size_t *offset)
+{
+    PyObject *ret = NULL;
+    PyObject *array = NULL;
+    PyArray_StringDTypeObject *string_dtype = NULL;
+    npy_string_allocator *allocator = NULL;
+    char *array_data = NULL;
+    npy_intp dims[1];
+    tsk_size_t i;
+    int pack_result;
+
+    string_dtype = (PyArray_StringDTypeObject *) PyArray_DescrFromType(NPY_VSTRING);
+    if (string_dtype == NULL) {
+        goto out;
+    }
+    dims[0] = (npy_intp) num_rows;
+    array = PyArray_Zeros(1, dims, (PyArray_Descr *) string_dtype, 0);
+    if (array == NULL) {
+        goto out;
+    }
+    allocator = NpyString_acquire_allocator(string_dtype);
+    if (allocator == NULL) {
+        goto out;
+    }
+    array_data = (char *) PyArray_DATA((PyArrayObject *) array);
+    for (i = 0; i < num_rows; i++) {
+        pack_result = NpyString_pack(allocator,
+            (npy_packed_static_string
+                    *) (array_data + (i * ((PyArray_Descr *) string_dtype)->elsize)),
+            data + offset[i], offset[i + 1] - offset[i]);
+        if (pack_result == -1) {
+            goto out;
+        }
+    }
+
+    if (PyArray_SetBaseObject((PyArrayObject *) array, (PyObject *) self) != 0) {
```

Comment:
Why are we setting the base object and increffing self here? I thought the allocator was just creating a new copy of the string data so there's no need for this?